src/include/duckdb/optimizer/optimizer_extension.hpp

```diff
@@ -13,14 +13,22 @@
 
 namespace duckdb {
 
+class Optimizer;
+class ClientContext;
+
 //! The OptimizerExtensionInfo holds static information relevant to the optimizer extension
 struct OptimizerExtensionInfo {
 	virtual ~OptimizerExtensionInfo() {
 	}
 };
 
-typedef void (*optimize_function_t)(ClientContext &context, OptimizerExtensionInfo *info,
-                                    unique_ptr<LogicalOperator> &plan);
+struct OptimizerExtensionInput {
+	ClientContext &context;
```

Comment:
I agree, originally I just included the `optimizer` but the idea with passing it directly was to avoid having every optimize extension need to include the `optimizer.hpp` header (in case you don't need it but do need the client context), or create a new `.cpp` file to hide the implementation.